srml/support/src/storage/generator/linked_map.rs

```diff
@@ -151,22 +159,32 @@ where
 
 	if let Some(prev_key) = prev_key {
 		// Retrieve previous element and update `next`
-		let mut res = read_with_linkage::<_, _, G>(prev_key.as_ref())
-			.expect("Linkage is updated in case entry is removed;
-					it always points to existing keys; qed");
-		res.1.next = linkage.next;
-		unhashed::put(prev_key.as_ref(), &res);
+		if let Some(mut res) = read_with_linkage::<_, _, G>(prev_key.as_ref()) {
+			res.1.next = linkage.next;
+			unhashed::put(prev_key.as_ref(), &res);
+		} else {
+			// TODO #3700: error should be handleable.
+			runtime_print::print(
+				"ERROR: Corrupted state: linked map {}: previous value doesn't exist at {:x}",
```

Comment:
We need to call into the else branch below to write the correct head.

Reply:
Yes, I was mixed about what was the best and conclude the least we touch things the best it is.

Here either:
* we let the head where it is, knowing that the list has been cut in two, with second part having a wrong head.
* move the head to next values potentially cutting a large part of the list.

Reply:
But this means we will print out this message over and over again. (Given there is no modifcation in between)

Reply:
I don't think this code should attempt to fix stuff, but ok maybe we can do it.

Then we should also maybe:
* remove entry when in storage::get decode has failing, 
* set the next to None when enumeration is broken,

Reply:
oh ya ok at least we can make previous being None for the next element. indeed